Dockerfile for methods VM

mike-w-wilson opened the pull request:
This contains the Dockerfile for the methods VM. Tools installed include htslib, samtools, bcftools, picard, vcftools, and GATK. Python and R are also installed as well as the hail library.

bw2 commented on vm-docker/Dockerfile:
```diff
@@ -0,0 +1,117 @@
+	# clean up apt cache
+    rm -rf /var/lib/apt/lists/*
+
+RUN mkdir home/tools 
```
/home/tools seems like an unconventional location since 'tools' isn't a user account. 
`/tools` or `/usr/tools` or `/home/root/tools` might be better.

mike-w-wilson replied:
Thanks Ben, I didnt think of this! I'm going with `/tools` for now but do not have a strong preference.

bw2 commented on vm-docker/Dockerfile:
```diff
@@ -0,0 +1,117 @@
+    libssl-dev \
+    openssl \
+    python-smbus \
+    python3 python-pip 
```
would split with \ like previous lines

bw2 commented on vm-docker/Dockerfile:
```diff
@@ -0,0 +1,117 @@
+RUN apt-get update \
+	&& apt-get dist-upgrade -y \
+	&& apt-get install -y \
+    g++ \
```
g++ already installed above

mike-w-wilson replied:
Good catch!

bw2 commented on vm-docker/Dockerfile:
```diff
@@ -0,0 +1,117 @@
+# Install python packages
+RUN apt-get update \
+	&& apt-get dist-upgrade -y \
+	&& apt-get install -y \
```
Would add --no-install-recommends here and other places
(from tip 4 @ https://www.docker.com/blog/intro-guide-to-dockerfile-best-practices/)

bw2 commented on vm-docker/Dockerfile:
```diff
@@ -0,0 +1,117 @@
+RUN apt-get install -y \
+    r-base=${R_VERSION} \
+    r-base-dev=${R_VERSION} 
+
```
Depending on how this image will be used, may want to also pre-install `bedtools` (https://github.com/broadinstitute/str-callers/blob/master/docker/base-other-tools/Dockerfile#L17-L26) 
and python libs like 
`scipy numpy pandas matplotlib seaborn ipython pybedtools`

mike-w-wilson replied:
I'll add bedtools and those python libraries - they all make sense to me and I believe are on many peoples init scripts.

bw2 commented on vm-docker/Dockerfile:
```diff
@@ -0,0 +1,117 @@
+	# clean up apt cache
+    rm -rf /var/lib/apt/lists/*
+
+RUN mkdir home/tools 
```
use `WORKDIR /home/tools` ( https://docs.docker.com/engine/reference/builder/#workdir )
to avoid `cd home/tools` in every RUN

mike-w-wilson replied:
Great tip!

bw2 reviewed (approved):
Added some suggestions, but looks good overall.

mike-w-wilson commented:
Hey @bw2, I made the changes you requested but also updated the Java 8 install for hail. It is a bit hacky because of security issues around `openjdk-8-jdk` on Debian but the way I implemented it was the only solution I could find. Please let me know if you think it is alright or if there are alternative solutions.

bw2 commented on vm-docker/Dockerfile:
```diff
@@ -0,0 +1,141 @@
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    apt-utils \
+    default-jdk \
```
JDK 8 install below looks fine to me.  Would get rid of default-jdk here.

mike-w-wilson commented:
Sorry @bw2 , looks like this repo dismisses old approvals after a new commit.  Would you mind approving again?
